### ISSUE TITLE
[dev.icinga.com #11149] host groups not exposed in object list when assigned via apply

### ISSUE THREAD

**Issue by icinga-migration** [Icinga/icinga2#3934]
This issue has been migrated from Redmine: https://dev.icinga.com/issues/11149

**Created by ricardo on 2016-02-11 20:29:28 +00:00**

Assignee: _(none)_
Status: _New_ 
Target Version: _Backlog_
Last Update: _2016-06-01 16:41:55 +00:00 (in Redmine)_

    Icinga Version: 2.4.1
    Backport?: Not yet backported
    Include in Changelog: 1

---

Hi,

if a host group is defined via apply then icinga object list does not show the assignment.

objects.conf

    object HostGroup "gentoo-servers" {
      display_name = "Gentoo Servers"

      assign where host.vars.os == "gentoo"
    }

icinga2 object list --type Host --name webserver

    bject 'webserver' of type 'Host':
      * __name = "webserver"
      * action_url = "/pnp4nagios/index.php/graph?host=webserver&srv=_HOST_&version=tiny&keepThis=true&TB_iframe=true&height=600&width=620' class='pnp4nagios'"
      * address = "10.8.0.222"
      * address6 = ""
      * check_command = "hostalive"
      * check_interval = 60
      * check_period = ""
      * command_endpoint = ""
      * display_name = "webserver"
      * enable_active_checks = true
      * enable_event_handler = true
      * enable_flapping = false
      * enable_notifications = true
      * enable_passive_checks = true
      * enable_perfdata = true
      * event_command = ""
      * flapping_threshold = 30
      * groups = [ ]
      * icon_image = "vendors/gentoo.png"
      * icon_image_alt = "Gentoo Linux"
      * max_check_attempts = 3
      * name = "webserver"
      * notes = ""
      * notes_url = ""
      * package = "_etc"
      * retry_interval = 30
      * templates = [ "webserver", "generic-host" ]
      * type = "Host"
      * vars
        * os = "gentoo"
      * volatile = false
      * zone = ""

via api:

    {
        __name = "webserver"
        acknowledgement = 0.000000
        acknowledgement_expiry = 0.000000
        action_url = "/pnp4nagios/index.php/graph?host=webserver&srv=_HOST_&version=tiny&keepThis=true&TB_iframe=true&height=600&width=620' class='pnp4nagios'"
        active = true
        address = "10.8.0.222"
        address6 = ""
        check_attempt = 1.000000
        check_command = "hostalive"
        check_interval = 60.000000
        check_period = ""
        command_endpoint = ""
        display_name = "webserver"
        enable_active_checks = true
        enable_event_handler = true
        enable_flapping = false
        enable_notifications = true
        enable_passive_checks = true
        enable_perfdata = true
        event_command = ""
        extensions = null
        flapping = false
        flapping_last_change = 1455222392.019179
        flapping_negative = 1626.000000
        flapping_positive = 0.000000
        flapping_threshold = 30.000000
        force_next_check = false
        force_next_notification = false
        groups = [ "gentoo-servers" ]
        ha_mode = 0.000000
        icon_image = "vendors/gentoo.png"
        icon_image_alt = "Gentoo Linux"
        last_check = 1455222392.019158
        last_check_result = {
            active = true
            check_source = "internetausdrucker"
            command = [ "/opt/icinga2/libexec/check_ping", "-H", "10.8.0.222", "-c", "5000,100%", "-w", "3000,80%" ]
            execution_end = 1455222392.019097
            execution_start = 1455222388.011432
            exit_status = 0.000000
            output = "PING OK - Packet loss = 0%, RTA = 0.31 ms"
            performance_data = [ "rta=0.311000ms;3000.000000;5000.000000;0.000000", "pl=0%;80;100;0" ]
            schedule_end = 1455222392.019158
            schedule_start = 1455222448.010000
            state = 0.000000
            type = "CheckResult"
            vars_after = {
                attempt = 1.000000
                reachable = true
                state = 0.000000
                state_type = 1.000000
            }
            vars_before = {
                attempt = 1.000000
                reachable = true
                state = 0.000000
                state_type = 1.000000
            }
        }
        last_hard_state = 0.000000
        last_hard_state_change = 1454128333.432491
        last_hard_state_raw = 0.000000
        last_in_downtime = false
        last_reachable = true
        last_state = 0.000000
        last_state_change = 1454128333.432491
        last_state_down = 1454128269.754376
        last_state_raw = 0.000000
        last_state_type = 1.000000
        last_state_unreachable = 0.000000
        last_state_up = 1455222392.019173
        max_check_attempts = 3.000000
        name = "webserver"
        next_check = 1455222448.010000
        notes = ""
        notes_url = ""
        original_attributes = null
        package = "_etc"
        pause_called = false
        paused = false
        resume_called = true
        retry_interval = 30.000000
        start_called = true
        state = 0.000000
        state_loaded = true
        state_raw = 0.000000
        state_type = 1.000000
        stop_called = false
        templates = [ "webserver", "generic-host" ]
        type = "Host"
        vars = {
            os = "gentoo"
        }
        version = 0.000000
        volatile = false
        zone = ""
    }

---

**Relations**:

* relates [#11149](https://dev.icinga.com/issues/11149)

**Comment by dnsmichi** [comment]
@plarivee If you modify a custom attribute at runtime, apply rules are not re-evaluated (that's #4084).

**Comment by dnsmichi** [comment]
That's explicitly mentioned in the docs as part of #5140.

**Comment by dnsmichi** [comment]
There's no way we can fix this, unless we skip the stream writing of the debug file and just dump everything in the end. I don't think this performance loss is worth the info, also since it is now a documented behaviour.
